src/runtime/index.ts

```diff
@@ -1,7 +1,123 @@
-import platform from "./platform";
+import platform, { Platform } from "./platform";
+import { EventEmitter } from "node:events";
+import { Transport } from "./transport";
+import { MessageEvent } from "../proto/server/v1/realtime";
 
-export class Realtime {
+type SubscribeCallback = (string) => void;
+
+export interface RealTimeConfig {
+  /*
+   * An id used to identify this client when using the presence feature.
+   * If the client id is not set, an error will be thrown when using any presence features
+   */
+  clientId?: string;
+  platform: Platform;
+}
+
+export class RealTime {
+  private _config: RealTimeConfig;
+  private channelManager: ChannelManager;
+  private transport: Transport;
+  constructor(config: RealTimeConfig) {
+    this.transport = new Transport({
+      WebSocket: config.platform.WebSocket,
+      heartbeatTimeout: 1000,
+    });
+    this.channelManager = new ChannelManager(this.transport);
+
+    this._config = config;
+  }
+
+  async connect() {
+    await this.transport.isConnected();
+  }
+
+  getChannel(name: string): Channel {
+    return this.channelManager.getOrCreate(name);
+  }
+
+  close() {
+    this.channelManager.close();
+    this.transport.close();
+  }
+
+  socketId() {
+    return this.transport.socketId();
+  }
+}
+
+class Presence extends EventEmitter {
   constructor() {
-    console.log("constructor", platform.runtime, platform.WebSocket);
+    super();
+  }
+}
+
+export class Channel extends EventEmitter {
+  private name: string;
+  private hasAttached: boolean;
+  private transport: Transport;
+
+  constructor(name: string, transport: Transport) {
```

Comment:
we also need to have ChannelOptions so that if we introduce anything on the channel level then that can be moved to those options.

Reply:
Do we have any current channel options? Otherwise I'll add that once we need to

Reply:
We can add it later as long as user-facing APIs don't change.